test/client/presence/doc-presence.js

```diff
@@ -705,7 +705,13 @@ describe('DocPresence', function() {
           });
         });
 
-        doc1.submitOp({index: 5, value: 'ern'}, errorHandler(done));
+        doc1.submitOp({index: 5, value: 'ern'}, errorHandler(function(error) {
+          if (!error) {
+            return done('should throw an error');
+          }
```

Comment:
Is this here because `mocha` swallows assertion errors inside callbacks?

If that's the case, can we write a little helper to tidy this pattern up? Something like:

```js
function asyncAssert(done, cb) {
  return function() {
    try {
      cb.apply(this, arguments);
    } catch (error) {
      done(error);
    }
  }
}

doc1.submitOp({...}, asyncAssert(done, function(error) {
  expect(error.code).to.be.equal('ERR_OP_SUBMIT_REJECTED');
});
```

Reply:
I believe Mocha does handle assertion errors inside callbacks properly, so this separate `if (!error)` check may not be needed. Try removing and see if it's ok?